bottlecap/src/lifecycle/invocation/triggers/step_function_event.rs

```diff
@@ -118,16 +131,36 @@ impl Trigger for StepFunctionEvent {
 impl StepFunctionEvent {
     #[must_use]
     pub fn get_span_context(&self) -> SpanContext {
-        let (lo_tid, hi_tid) = Self::generate_trace_id(self.execution.id.clone());
-        let tags = HashMap::from([(
-            DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY.to_string(),
-            format!("{hi_tid:x}"),
-        )]);
+        let (lo_tid, tags) =
+            if let (Some(trace_id), Some(trace_tags)) = (&self.trace_id, &self.trace_tags) {
+                // Lambda Root
+                let lo_tid = trace_id.parse().expect("Failed to parse trace_id to u64");
```

Comment:
Is there a way to avoid using `expect` here?

Reply:
refactored so we use `Self::generate_trace_id` as a fallback if the parse fails